File: features/transaction/checkout/src/test/java/com/tokopedia/checkout/domain/mapper/ShipmentMapperTest.java
package com.tokopedia.checkout.domain.mapper;

import com.google.gson.Gson;
import com.tokopedia.UnitTestFileUtils;
import com.tokopedia.checkout.domain.datamodel.cartshipmentform.CartShipmentAddressFormData;
import com.tokopedia.transactiondata.entity.response.shippingaddressform.ShipmentAddressFormDataResponse;

import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

/**
 * Created by Fajar Ulin Nuha on 15/11/18.
 */
public class ShipmentMapperTest {

    private static final String PATH_CONTRACT_PURCHASE_PROTECTION = "assets/contract_purchase_protection.json";
    private static final String PATH_CONTRACT_NULL_PURCHASE_PROTECTION = "assets/contract_purchase_protection_null.json";

    private UnitTestFileUtils unitTestFileUtils;
    private Gson gson;
    private ShipmentMapper shipmentMapper;

    @Before
    public void setUp() {
        MapperUtil mapperUtil = new MapperUtil();
        unitTestFileUtils = new UnitTestFileUtils();
        gson = new Gson();
        shipmentMapper = new ShipmentMapper(mapperUtil);
    }


    @Test
    public void convertToShipmentAddressFormDataWithContractPurchaseProtectionNoException() {
        // given
        ShipmentAddressFormDataResponse response =
                gson.fromJson(unitTestFileUtils.getJsonFromAsset(PATH_CONTRACT_PURCHASE_PROTECTION),
                        ShipmentAddressFormDataResponse.class);

        // when
        CartShipmentAddressFormData result = shipmentMapper.convertToShipmentAddressFormData(response);

        // verify
        Assert.assertNotNull(result.getGroupAddress().get(0).getGroupShop()
                .get(0).getProducts().get(0).getPurchaseProtectionPlanData());
    }

    @Test
    public void convertToShipmentAddressFormDataWithContractNullPurchaseProtectionNoException() {
        // given
        ShipmentAddressFormDataResponse response =
                gson.fromJson(unitTestFileUtils.getJsonFromAsset(PATH_CONTRACT_NULL_PURCHASE_PROTECTION),
                        ShipmentAddressFormDataResponse.class);

        // when
        CartShipmentAddressFormData result = shipmentMapper.convertToShipmentAddressFormData(response);

        // verify
        Assert.assertNull(result.getGroupAddress().get(0).getGroupShop()
                .get(0).getProducts().get(0).getPurchaseProtectionPlanData());
    }
}

File: features/transaction/checkout/src/test/java/com/tokopedia/checkout/domain/mapper/ShipmentMapperTest.kt
package com.tokopedia.checkout.domain.mapper

import com.google.gson.Gson
import com.tokopedia.checkout.UnitTestFileUtils
import com.tokopedia.checkout.data.model.response.shipmentaddressform.ShipmentAddressFormDataResponse
import org.junit.Assert.assertNotNull
import org.junit.Assert.assertNull
import org.junit.Before
import org.junit.Test

/**
 * Created by Fajar Ulin Nuha on 15/11/18.
 */
class ShipmentMapperTest {

    private var unitTestFileUtils: UnitTestFileUtils? = null
    private var gson: Gson? = null
    private var shipmentMapper: ShipmentMapper? = null

    @Before
    fun setUp() {
        unitTestFileUtils = UnitTestFileUtils()
        gson = Gson()
        shipmentMapper = ShipmentMapper()
    }

    @Test
    fun convertToShipmentAddressFormDataWithContractPurchaseProtectionNoException() {
        // given
        val response = gson?.fromJson(unitTestFileUtils?.getJsonFromAsset(PATH_CONTRACT_PURCHASE_PROTECTION),
                ShipmentAddressFormDataResponse::class.java) ?: ShipmentAddressFormDataResponse()

        // when
        val cartShipmentAddressFormData = shipmentMapper?.convertToShipmentAddressFormData(response)

        // verify
        assertNotNull(cartShipmentAddressFormData?.groupAddress?.get(0)?.groupShop?.get(0)?.products?.get(0)?.purchaseProtectionPlanData)
    }

    companion object {
        private const val PATH_CONTRACT_PURCHASE_PROTECTION = "assets/contract_purchase_protection.json"
    }
}